File: inJava/app/src/main/java/com/barrygithub/rickandmortyapp/domain/UseCaseInteractor.java
package com.barrygithub.rickandmortyapp.domain;

import com.barrygithub.rickandmortyapp.data.LocalDatasource.entities.EntityDb;

import io.reactivex.rxjava3.core.Observable;

/**
 * Project RickAndMortyApp
 * Created by Barry Zea H. on 7/04/23.
 * Copyright (c) Barry Zea H. All rights reserved.
 **/
public interface  UseCaseInteractor {
    void getData(int page);
    Observable<EntityDb> observerData();

}

File: inKotlin/app/src/main/java/com/barrygithub/rickandmortyapp/domain/UseCaseInteractor.kt
package com.barrygithub.rickandmortyapp.domain

import com.barrygithub.rickandmortyapp.data.localDatasource.EntityDb
import io.reactivex.rxjava3.subjects.BehaviorSubject

/**
 * Project RickAndMortyApp
 * Created by Barry Zea H. on 14/04/23.
 * Copyright (c) Barry Zea H. All rights reserved.
 *
 **/
interface UseCaseInteractor {
    fun getObservableData():BehaviorSubject<EntityDb>
    fun getDataGraphql(page:Int)
}